src/steps/synchronize/index.test.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  executeStepWithDependencies,
+  Recording,
+} from '@jupiterone/integration-sdk-testing';
+
+import { setupCiscoAmpRecording } from '../../../test/recording';
+import { buildStepTestConfigForStep } from '../../../test/config';
+import {
+  Entities,
+  MappedRelationships,
+  Relationships,
+  Steps,
+} from '../constants';
+
+describe('syncronizeStep', () => {
+  let recording: Recording;
+  afterEach(async () => {
+    await recording.stop();
+  });
+
+  describe('#fetchComputers', () => {
+    test('should create account and computer entities and relationships', async () => {
+      recording = setupCiscoAmpRecording({
+        name: 'cisco_amp',
+        directory: __dirname,
+        redactedRequestHeaders: ['api-key'],
+        options: {
+          recordFailedRequests: false,
+          matchRequestsBy: {
+            url: {
+              query: false,
+            },
+          },
+        },
+      });
+
+      const stepConfig = buildStepTestConfigForStep(Steps.SYNCHRONIZE);
+      const stepResult = await executeStepWithDependencies(stepConfig);
+
+      const accountEntity = stepResult.collectedEntities.filter(
+        (e) => e._type === Entities.ACCOUNT._type,
+      );
+      const computerEntities = stepResult.collectedEntities.filter(
+        (e) => e._type === Entities.COMPUTER._type,
+      );
+      const accountComputerRelationships = stepResult.collectedRelationships.filter(
+        (e) => e._type === Relationships.ACCOUNT_HAS_ENDPOINT._type,
+      );
+
+      const endpointProtectsDeviceMappedRelationships = stepResult.collectedRelationships.filter(
+        (e) => e._type === MappedRelationships.ENDPOINT_PROTECTS_DEVICE._type,
+      );
+
+      // Assert that all collected objects have length >= 1
+      expect(accountEntity.length).toBe(1);
+      expect(computerEntities.length).toBeGreaterThanOrEqual(1);
+      expect(accountComputerRelationships.length).toBeGreaterThanOrEqual(1);
+      expect(
+        endpointProtectsDeviceMappedRelationships.length,
+      ).toBeGreaterThanOrEqual(1);
+
+      // Asset they match schema
+      expect(accountEntity).toMatchGraphObjectSchema(Entities.ACCOUNT);
+      expect(computerEntities).toMatchGraphObjectSchema(Entities.COMPUTER);
+      expect(accountComputerRelationships).toMatchDirectRelationshipSchema(
+        Relationships.ACCOUNT_HAS_ENDPOINT,
+      );
+
+      // There isn't a matcher for mapped relationships in the sdk, so
+      // for now this is a simple manual check
+      endpointProtectsDeviceMappedRelationships.map((rel) => {
```

Comment:
Note that you can use `toTargetEntities` to ensure each specific user_endpoint is being mapped to.
example: https://github.com/JupiterOne/graph-veracode-v2/blob/main/src/steps/findings/index.test.ts#L63

Reply:
It looks like `toTargetEntities` doesn't work with `targetFilterKeys` that are Arrays. This may or may not be a bug, but since it's already the existing behavior, I'm going to merge this and come back to it when I've found out if it's wrong to use an Array in `targetFilterKeys` or if `toTargetEntities` is bugged.